### DISCUSSION TITLE
Releasing an AddOn that depends on gdUnit4

### DISCUSSION THREAD

**Discussion by SamuelAsherRivello** [MikeSchulze/gdUnit4#521] [General]
I'm a game dev and unit testing expert. 

https://www.samuelasherrivello.com/

I'm new to Godot and have questions about your great plugin.

# Summary

I want to offer my own addon to the Godot community. I want GDUnit4 tests to be included within. Ideally end users can run the tests.

# Questions

1. Does godot asset library support that one addon depend on another addon?
2. If not, is there another way to deliver addons that offers dependency support (e.g. npm) that is already popular in the community?
3. Or I could include tests and perhaps 'turn them off' if the user does not have GDUnit4 installed. Using a #ifdef perhaps. Ideas?

**Comment by MikeSchulze** [comment]
> Does godot asset library support that one addon depend on another addon?

Unfortunately no, the plugin support in Godot is quite limited.
- it does not support dependencies
- it does not support update mechanisms
- it has no hot reload of plugin during developing
- it has actual a lot of issues at startup to load scripts using `class_name`

> If not, is there another way to deliver addons that offers dependency support (e.g. npm) that is already popular in the community?

I can't answer that, in the best case there is already a plugin that manages dependencies.
Otherwise, take a look at the feature request in Godot GitHub

> Or I could include tests and perhaps 'turn them off' if the user does not have GDUnit4 installed. Using a #ifdef perhaps. Ideas?

If no GdUnit4 plugin installed it not a problem, the test will just have no effect.
If you want to integrate a CI workflow for automatic test run on your pull request checkout [gdUnit4-action](https://github.com/MikeSchulze/gdUnit4-action)

**Comment by SamuelAsherRivello** [comment]
1. That is unfortunate. Thanks.
2. Gotcha. In the unity world you can use a 3rd party like Npm.js to store libraries (that don't depend on unity classes) and have packages that depend on packages (as long as all are on Npm.js. Then use commandline (completely separately from Unity itself) to manage plugins. That offers some advantages but also disadvantages compared to the Unity-centric Package manager solution. Perhaps a workflow like that for Godot will be developed. Ultimately, a godot-centric (native not plugin) solution with such dependency management would be great.
3. Cool.
